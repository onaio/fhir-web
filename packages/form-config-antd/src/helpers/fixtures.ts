/* eslint-disable @typescript-eslint/camelcase */
export const manifestRelease4 = {
  identifier: '1.0.30',
  json: '{"forms_version":"1.0.16","identifiers":["blood_screening.json"}',
  appId: 'org.smartregister.foo',
  appVersion: '1.2.18',
  createdAt: '2020-06-09T14:40:04.189+02:00',
  updatedAt: '2020-06-09T14:40:04.189+02:00',
};

export const fixManifestReleases = [
  {
    identifier: '1.0.11',
    json: '{"forms_version":"1.0.11","identifiers":["blood_screening.json"}',
    appId: 'org.smartregister.reveal',
    appVersion: '1.2.14',
    createdAt: '2020-06-08T15:00:38.413+02:00',
    updatedAt: '2020-06-08T15:00:38.413+02:00',
  },
  {
    identifier: '1.0.12',
    json: '{"forms_version":"1.0.12","identifiers":["blood_screening.json"}',
    appId: 'org.smartregister.reveal',
    appVersion: '1.2.14',
    createdAt: '2020-06-08T15:03:00.950+02:00',
    updatedAt: '2020-06-08T15:03:00.950+02:00',
  },
  {
    identifier: '1.0.16',
    json: '{"forms_version":"1.0.16","identifiers":["blood_screening.json"}',
    appId: 'org.smartregister.reveal',
    appVersion: '1.2.16',
    createdAt: '2020-06-09T14:40:04.189+02:00',
    updatedAt: '2020-06-09T14:40:04.189+02:00',
  },
];

export const manifestFile1 = {
  createdAt: 'Jun 19, 2020, 12:31:37 PM',
  form_relation: '',
  id: '52',
  identifier: 'test-form-1.json',
  isDraft: false,
  isJsonValidator: false,
  jursdiction: '',
  label: 'test form',
  module: '',
  version: '1.0.26',
};

export const manifestFile2 = {
  createdAt: 'Jun 19, 2020, 4:23:22 PM',
  form_relation: '',
  id: '53',
  identifier: 'reveal-test-file.json',
  isDraft: false,
  isJsonValidator: false,
  jursdiction: '',
  label: 'test publish',
  module: '',
  version: '1.0.27',
};

export const manifestFile3 = {
  createdAt: 'Jun 19, 2020, 6:23:22 PM',
  form_relation: '',
  id: '54',
  identifier: 'foo.json',
  isDraft: false,
  isJsonValidator: false,
  jursdiction: '',
  label: 'foo',
  module: 'baz',
  version: '1.0.28',
};

export const draftFile1 = {
  ...manifestFile1,
  isDraft: true,
  isJsonValidator: false,
};

export const draftFile2 = {
  ...manifestFile2,
  isDraft: true,
  isJsonValidator: false,
};

export const draftFile3 = {
  ...manifestFile3,
  isDraft: true,
  isJsonValidator: false,
};

export const fixManifestFiles = [manifestFile1, manifestFile2];

export const FixManifestDraftFiles = [
  {
    ...fixManifestFiles[0],
    isDraft: false,
  },
  {
    ...fixManifestFiles[1],
    isDraft: false,
  },
];

export const downloadFile = {
  clientForm: {
    json: JSON.stringify(fixManifestFiles[0]),
  },
};
