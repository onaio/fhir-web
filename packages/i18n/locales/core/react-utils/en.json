{
  "404": "",
  "Actions": "",
  "Class": "",
  "Date created": "",
  "Error": "",
  "Fetching next page": "",
  "Go back": "",
  "Go home": "",
  "ID": "",
  "Load more options": "",
  "No data": "",
  "No data available": "",
  "Search": "",
  "Showing {{recordsFetchedNum}}; {{remainingRecords}} more records.": "",
  "Something went wrong": "",
  "Sorry, the resource you requested for, does not exist": "",
  "Sorry, you are not authorized to access this page": "",
  "status": "",
  "Unable to load dropdown options.": "",
  "View details": ""
}
