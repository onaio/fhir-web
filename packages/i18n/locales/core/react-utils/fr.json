{
  "403": "",
  "404": "",
  "Actions": "",
  "Error": "",
  "Go back": "",
  "Go home": "",
  "Search": "",
  "Something went wrong": "",
  "Sorry, the resource you requested for, does not exist": "",
  "Sorry, you are not authorized to access this page": ""
}
