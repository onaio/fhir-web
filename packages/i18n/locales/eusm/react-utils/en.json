{
  "403": "",
  "404": "",
  "Actions": "",
  "Error": "",
  "Fetching next page": "Fetching next page",
  "Go back": "",
  "Go home": "",
  "Load more options": "Load more options",
  "No data": "No data",
  "Search": "",
  "Showing {{recordsFetchedNum}}; {{remainingRecords}} more records.": "Showing {{recordsFetchedNum}}; {{remainingRecords}} more records.",
  "Something went wrong": "",
  "Sorry, the resource you requested for, does not exist": "",
  "Sorry, you are not authorized to access this page": "",
  "Unable to load dropdown options.": "Unable to load dropdown options."
}
