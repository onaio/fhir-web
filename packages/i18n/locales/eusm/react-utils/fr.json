{
  "403": "403",
  "404": "404",
  "Actions": "Actions",
  "Error": "Erreur",
  "Go back": "Retour",
  "Go home": "Retour à l'accueil",
  "Search": "Recherche",
  "Session Expired": "Session expirée",
  "Something went wrong": "Il y a eu un problème",
  "Sorry, the resource you requested for, does not exist": "Désolé, la ressource que vous avez demandée n'existe pas",
  "Sorry, you are not authorized to access this page": "Désolé, vous n'êtes pas autorisé à accéder à cette page"
}
