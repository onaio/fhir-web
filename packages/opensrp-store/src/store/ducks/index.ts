/** users ducks modules: actions, actionCreators, reducer and selectors */
import { Dictionary } from '@onaio/utils';
import { keyBy, values } from 'lodash';
import intersect from 'fast_array_intersect';
import { AnyAction, Store } from 'redux';
import { createSelector, OutputParametricSelector } from 'reselect';
import SeamlessImmutable from 'seamless-immutable';

/** The reducer name */
export const reducerName = 'keycloakUsers';

/** Interface for user json object */
export interface KeycloakUser {
  access?: {
    manageGroupMembership: boolean;
    view: boolean;
    mapRoles: boolean;
    impersonate: boolean;
    manage: boolean;
  };
  createdTimestamp?: number;
  disableableCredentialTypes?: string[];
  email?: string;
  emailVerified?: boolean;
  enabled?: boolean;
  firstName: string;
  id: string;
  lastName: string;
  notBefore?: number;
  requiredActions?: string[];
  totp?: boolean;
  username: string;
}

// actions

/** action type for fetching keycloak users */
export const KEYCLOAK_USERS_FETCHED = 'keycloak/reducer/users/userS_FETCHED';
/** action type for removing keycloak users */
export const REMOVE_KEYCLOAK_USERS = 'keycloak/reducer/users/REMOVE_KEYCLOAK_USERS';

/** interface action to add users to store */
export interface FetchKeycloakUsersAction extends AnyAction {
  usersById: Dictionary<KeycloakUser>;
  type: typeof KEYCLOAK_USERS_FETCHED;
}

/** Interface for removeusersAction */
export interface RemoveKeycloakUsersAction extends AnyAction {
  usersById: Dictionary<KeycloakUser>;
  type: typeof REMOVE_KEYCLOAK_USERS;
}

/** Create type for users reducer actions */
export type KeycloakUsersActionTypes =
  | FetchKeycloakUsersAction
  | RemoveKeycloakUsersAction
  | AnyAction;

// action Creators

/**
 * Fetch users action creator
 *
 * @param {Array} usersList - keycloak users
 * @returns {object} - the dispatcher to remove the user
 */
export const fetchKeycloakUsers = (usersList: KeycloakUser[] = []): FetchKeycloakUsersAction => {
  return {
    usersById: keyBy(usersList, (user: KeycloakUser) => user.id),
    type: KEYCLOAK_USERS_FETCHED,
  };
};

/** Remove users action creator
 *
 * @returns {object} - the dispatcher to remove the user
 */
export const removeKeycloakUsers = (): RemoveKeycloakUsersAction => {
  return {
    usersById: {},
    type: REMOVE_KEYCLOAK_USERS,
  };
};

// The reducer

/** interface for keycloak users state in redux store */
export interface KeycloakUsersState {
  usersById: Dictionary<KeycloakUser> | Dictionary;
}

/** Create an immutable keycloak users state */
export type ImmutableKeycloakUsersState = KeycloakUsersState &
  SeamlessImmutable.ImmutableObject<KeycloakUsersState>;

/** initial keycloak users state */
export const initialState: ImmutableKeycloakUsersState = SeamlessImmutable({
  usersById: {},
});

/**
 * the users reducer function
 *
 * @param {object} state - keycloak users states
 * @param {object} action - keycloak users actions
 * @returns {object} - the updated states
 */
export function reducer(
  state: ImmutableKeycloakUsersState = initialState,
  action: KeycloakUsersActionTypes
): ImmutableKeycloakUsersState {
  switch (action.type) {
    case KEYCLOAK_USERS_FETCHED:
      return SeamlessImmutable({
        ...state,
        usersById: { ...state.usersById, ...action.usersById },
      });
    case REMOVE_KEYCLOAK_USERS:
      return SeamlessImmutable({
        ...state,
        usersById: action.usersById,
      });

    default:
      return state;
  }
}

// Selectors

export interface KeycloakUsersFilters {
  id: string[] /** get all users whose ids appear in this array */;
  username?: string /** get users whose username includes text in the username field */;
}

// NON MEMOIZED SELECTORS

/**
 * Gets id from KeycloakUsersFilters
 *
 * @param {any} _ - the redux store
 * @param {Array} props - the keycloak users filters object
 * @returns {string} - the keycloak user id
 */
export const getUserIds = (_: Partial<Store>, props: KeycloakUsersFilters): string[] => props.id;

/**
 * Gets name from KeycloakUsersFilters
 *
 * @param {any} _ - the redux store
 * @param {object} props - the users filters object
 * @returns {string} - the keycloak user name
 */
export const getUsername = (_: Partial<Store>, props: KeycloakUsersFilters): string | undefined =>
  props.username;

/** returns all users in the store as values whose keys are their respective ids
 *
 * @param {any} state - the redux store
 * @returns {object} - users object as values, respective ids as keys
 */
export function getKeycloakUsersById(state: Partial<Store>): { [key: string]: KeycloakUser } {
  return (state as Dictionary)[reducerName].usersById;
}

/** gets keycloak users as an array of user objects
 *
 * @param {any} state - the redux store
 * @returns {Array} - an array of users objs
 */
export function getKeycloakUsersArray(state: Partial<Store>): KeycloakUser[] {
  return values(getKeycloakUsersById(state));
}

/**
 * Gets all practitioners whose name includes phrase given in name filter prop
 *
 * @returns {Array} - practitioners whose name includes phrase given in name filter prop
 */
export const getUsersByUsername = (): OutputParametricSelector<
  Partial<Store>,
  KeycloakUsersFilters,
  KeycloakUser[],
  (res1: KeycloakUser[], res2: string | undefined) => KeycloakUser[]
> =>
  createSelector(getKeycloakUsersArray, getUsername, (usersArray, username) =>
    username
      ? usersArray.filter((user: KeycloakUser) =>
          user.username.toLowerCase().includes(username.toLowerCase())
        )
      : usersArray
  );

// MEMOIZED SELECTORS

/**
 * Gets all users whose identifiers appear in ids filter prop value
 *
 * @returns {Array} - users whose identifiers appear in ids filter prop value
 */
export const getKeycloakUsersByIds = (): OutputParametricSelector<
  Partial<Store>,
  KeycloakUsersFilters,
  KeycloakUser[],
  (
    res1: {
      [key: string]: KeycloakUser;
    },
    res2: string[],
    res3: KeycloakUser[]
  ) => KeycloakUser[]
> =>
  createSelector(
    getKeycloakUsersById,
    getUserIds,
    getKeycloakUsersArray,
    (usersById, ids, usersArray) => {
      if (ids === undefined) {
        return usersArray;
      }
      if (ids.length > 0) {
        return ids.map((id: string) => usersById[id]);
      }
      return [];
    }
  );

/** practitioner array selector factory
 * aggregates response from all applied filters and returns results
 *
 * @returns {Array} - aggregates response from all applied filters and returns results
 */
export const makeKeycloakUsersSelector = (): OutputParametricSelector<
  Partial<Store>,
  KeycloakUsersFilters,
  KeycloakUser[],
  (res1: KeycloakUser[], res2: KeycloakUser[]) => KeycloakUser[]
> =>
  createSelector(getKeycloakUsersByIds(), getUsersByUsername(), (arr1, arr2) => {
    return intersect([arr1, arr2], JSON.stringify);
  });
