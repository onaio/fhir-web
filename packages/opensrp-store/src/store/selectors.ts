import { reducerName as sessionReducerName } from '@onaio/session-reducer';
import { Store } from 'redux';
import { Dictionary } from '@onaio/utils';
import { createSelector } from 'reselect';

/** get API Token from the Redux store
 *
 * @param {any} state - the redux store
 */
interface StoreState {
  session: {
    [key: string]: Dictionary;
  };
}

interface APIFilters {
  accessToken?: boolean;
  apiToken?: boolean;
  providerState?: boolean;
}

/** Fetch access token from store if access_token filter val is true
 *
 * @param {object} _ - redux store
 * @param {object} props - api filters object
 * @returns {boolean} returns boolean value
 */
export const fetchAccessToken = (_: Partial<Store>, props: APIFilters): boolean | null =>
  props.accessToken || null;

/** Fetch api token from store if api_token filter val is true
 *
 * @param {object} _ redux store
 * @param {object} props api filters object
 * @returns {boolean} returns boolean value
 */
export const fetchApiToken = (_: Partial<Store>, props: APIFilters): boolean | null =>
  props.apiToken || null;

/** Fetch oauth provider state from store if providerState filter val is true
 *
 * @param {Object} _ redux store
 * @param {Object} props api filters object
 * @returns {boolean} returns boolean value
 */
export const fetchOauthProviderState = (_: Partial<Store>, props: APIFilters): boolean | null =>
  props.providerState || null;

/** Gets extra data object from store
 *
 * @param {object} state - redux store
 * @returns {Object}
 */

export const getExtraData = (state: Partial<Store>): Dictionary =>
  (state as StoreState)[sessionReducerName].extraData;

/** API state selector
 *
 * @returns {Object} - the states
 */
export const makeAPIStateSelector = () =>
  createSelector(
    [fetchAccessToken, fetchApiToken, fetchOauthProviderState, getExtraData],
    (accessToken, apiToken, oauthState, extraData) => {
      if (accessToken) {
        return (extraData.oAuth2Data && extraData.oAuth2Data.access_token) || null;
      } else if (apiToken) {
        return extraData.api_token || null;
      } else if (oauthState) {
        return (extraData.oAuth2Data && extraData.oAuth2Data.state) || null;
      }
    }
  );
