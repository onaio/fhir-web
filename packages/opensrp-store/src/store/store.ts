import { connectReducer, getConnectedStore } from '@onaio/connected-reducer-registry';
import { gateKeeperReducer, gateKeeperReducerName } from '@onaio/gatekeeper';
import reducerRegistry, { combine, Registry } from '@onaio/redux-reducer-registry';
import session, { reducerName as sessionReducer } from '@onaio/session-reducer';
import { Reducer } from 'redux';
import { RouterState } from 'connected-react-router';
import { Dictionary } from '@onaio/utils';

/** declare globals interface */
declare global {
  interface Window {
    __PRELOADED_STATE__: Dictionary;
  }
}

/** Initial reducers in the reducer registry */
const defaultReducers: Registry = {
  router: connectReducer as Reducer<RouterState>,
};

// Grab the state from a global variable injected into the server-generated HTML
const preloadedState = window.__PRELOADED_STATE__ || {};
// Allow the passed state to be garbage-collected
delete window.__PRELOADED_STATE__;

/** Add users reducer to registry */
defaultReducers[sessionReducer] = session;
defaultReducers[gateKeeperReducerName] = gateKeeperReducer;

/** The initial store for the reveal web app */
export const store = getConnectedStore(defaultReducers, preloadedState);

/** Set listener to add reducers to store when registered */
reducerRegistry.setChangeListener((reducers) => {
  store.replaceReducer(combine(reducers));
});
