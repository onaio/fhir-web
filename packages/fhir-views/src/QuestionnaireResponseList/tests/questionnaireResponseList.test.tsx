import React from 'react';
import { store } from '@opensrp/store';
import { authenticateUser } from '@onaio/session-reducer';
import {
  fireEvent,
  render,
  screen,
  waitFor,
  waitForElementToBeRemoved,
} from '@testing-library/react';
import { createMemoryHistory } from 'history';
import { QuestionnaireResponseList } from '..';
import { QueryClientProvider, QueryClient } from 'react-query';
import { Router, Route, Switch } from 'react-router';
import nock from 'nock';
import { questionnairesPage1, questRespPage1, questRespPage2 } from '../../tests/fixtures';
import { RoleContext } from '@opensrp/rbac';
import { superUserRole } from '@opensrp/react-utils';
import { Provider } from 'react-redux';

jest.mock('fhirclient', () => {
  return jest.requireActual('fhirclient/lib/entry/browser');
});

const props = {
  fhirBaseURL: 'http://example.com',
};

const rQClient = new QueryClient();

// TODO - boiler plate
store.dispatch(
  authenticateUser(
    true,
    {
      email: 'bob@example.com',
      name: 'Bobbie',
      username: 'RobertBaratheon',
    },
    { api_token: 'hunter2', oAuth2Data: { access_token: 'bamboocha', state: 'abcde' } }
  )
);

// eslint-disable-next-line @typescript-eslint/no-explicit-any
const App = (props: any) => {
  return (
    <Provider store={store}>
      <Switch>
        <Route exact path="/qr/:id">
          <RoleContext.Provider value={superUserRole}>
            <QueryClientProvider client={rQClient}>{props.children}</QueryClientProvider>
          </RoleContext.Provider>
        </Route>
      </Switch>
    </Provider>
  );
};
nock.disableNetConnect();

test('pagination events work correctly', async () => {
  const history = createMemoryHistory();
  history.push('/qr/214');

  nock(props.fhirBaseURL)
    .get('/Questionnaire/214')
    .reply(200, questionnairesPage1.entry[0])
    .persist();

  nock(props.fhirBaseURL)
    .get('/QuestionnaireResponse/_search')
    .query({
      _total: 'accurate',
      _getpagesoffset: 0,
      _count: 20,
      questionnaire: '214',
    })
    .reply(200, questRespPage1)
    .persist();

  nock(props.fhirBaseURL)
    .get('/QuestionnaireResponse/_search')
    .query({
      _total: 'accurate',
      _getpagesoffset: 20,
      _count: 20,
      questionnaire: '214',
    })
    .reply(200, questRespPage2)
    .persist();
  nock(props.fhirBaseURL)
    .get('/QuestionnaireResponse/_search')
    .query({ _total: 'accurate', _getpagesoffset: 40, _count: 20, questionnaire: '214' })
    .reply(200, [])
    .persist();

  render(
    <Router history={history}>
      <App>
        <QuestionnaireResponseList {...props}></QuestionnaireResponseList>
      </App>
    </Router>
  );

  const waitForSpinner = async () => {
    return await waitFor(() => {
      expect(document.querySelector('.ant-spin')).not.toBeInTheDocument();
    });
  };

  await waitForElementToBeRemoved(document.querySelector('.ant-spin'));

  await waitFor(() => {
    expect(screen.getByText(/d8eb71c1-085d-4667-8fe1-b64ad1c6dd77/)).toBeInTheDocument();
  });

  document.querySelectorAll('tr').forEach((tr, idx) => {
    tr.querySelectorAll('td').forEach((td) => {
      expect(td).toMatchSnapshot(`table row ${idx} page 1`);
    });
  });

  fireEvent.click(screen.getByTitle('2'));

  expect(history.location.search).toEqual('?pageSize=20&page=2');

  await waitForSpinner();
  await waitForElementToBeRemoved(document.querySelector('.ant-spin'));

  expect(screen.getByText(/8d8d0c07-c2bf-4e5a-9cb3-6c264c3e58dd/)).toBeInTheDocument();
  document.querySelectorAll('tr').forEach((tr, idx) => {
    tr.querySelectorAll('td').forEach((td) => {
      expect(td).toMatchSnapshot(`table row ${idx} page 2`);
    });
  });
});
