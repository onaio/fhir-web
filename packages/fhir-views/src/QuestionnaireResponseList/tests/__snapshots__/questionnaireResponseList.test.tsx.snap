// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pagination events work correctly: table row 1 page 1 1`] = `
<td
  class="ant-table-cell"
>
  d8eb71c1-085d-4667-8fe1-b64ad1c6dd77
</td>
`;

exports[`pagination events work correctly: table row 1 page 1 2`] = `
<td
  class="ant-table-cell"
>
  12/30/2021
</td>
`;

exports[`pagination events work correctly: table row 1 page 1 3`] = `
<td
  class="ant-table-cell"
/>
`;

exports[`pagination events work correctly: table row 1 page 1 4`] = `
<td
  class="ant-table-cell"
>
  <a
    href="/quest-form/d8eb71c1-085d-4667-8fe1-b64ad1c6dd77/QuestionnaireResponse"
  >
    Edit
  </a>
</td>
`;

exports[`pagination events work correctly: table row 1 page 2 1`] = `
<td
  class="ant-table-cell"
>
  8d8d0c07-c2bf-4e5a-9cb3-6c264c3e58dd
</td>
`;

exports[`pagination events work correctly: table row 1 page 2 2`] = `
<td
  class="ant-table-cell"
>
  12/30/2021
</td>
`;

exports[`pagination events work correctly: table row 1 page 2 3`] = `
<td
  class="ant-table-cell"
/>
`;

exports[`pagination events work correctly: table row 1 page 2 4`] = `
<td
  class="ant-table-cell"
>
  <a
    href="/quest-form/8d8d0c07-c2bf-4e5a-9cb3-6c264c3e58dd/QuestionnaireResponse"
  >
    Edit
  </a>
</td>
`;

exports[`pagination events work correctly: table row 2 page 1 1`] = `
<td
  class="ant-table-cell"
>
  f93a4e65-d823-468c-842f-fd5a1509f231
</td>
`;

exports[`pagination events work correctly: table row 2 page 1 2`] = `
<td
  class="ant-table-cell"
>
  12/30/2021
</td>
`;

exports[`pagination events work correctly: table row 2 page 1 3`] = `
<td
  class="ant-table-cell"
/>
`;

exports[`pagination events work correctly: table row 2 page 1 4`] = `
<td
  class="ant-table-cell"
>
  <a
    href="/quest-form/f93a4e65-d823-468c-842f-fd5a1509f231/QuestionnaireResponse"
  >
    Edit
  </a>
</td>
`;

exports[`pagination events work correctly: table row 2 page 2 1`] = `
<td
  class="ant-table-cell"
>
  f8315582-38f9-4733-a932-2d2348a6af82
</td>
`;

exports[`pagination events work correctly: table row 2 page 2 2`] = `
<td
  class="ant-table-cell"
>
  12/29/2021
</td>
`;

exports[`pagination events work correctly: table row 2 page 2 3`] = `
<td
  class="ant-table-cell"
/>
`;

exports[`pagination events work correctly: table row 2 page 2 4`] = `
<td
  class="ant-table-cell"
>
  <a
    href="/quest-form/f8315582-38f9-4733-a932-2d2348a6af82/QuestionnaireResponse"
  >
    Edit
  </a>
</td>
`;
